### PR TITLE
Feature/107

### PR DESCRIPTION
**Pull request by rodriguez-facundo**
Data graph visualisation component.

Most of the logic it to center the graph with respect to Flexlayout or zoom the camera when we add OBJ files to the scene.

### CODE REVIEW

File: js/components/interface/graph-visualization/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        {source: 1, target: 2},
+        {source: 2, target: 3},
+        {source: 3, target: 1}
+    ]
+}
+```
+
+## With react-flexlayout
+
+If you are planning to use GGV with react-flexlayout, remember to add the resize event listener to the node where GGV will be attached. Example bellow:
```
Comment (filippomc) [inline js/components/interface/graph-visualization/README.md]: We probably don't need this with the suggested change (move logic to render)

File: js/components/interface/graph-visualization/forces/center.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+export default function (x, y, z) {
```
Comment (filippomc) [inline js/components/interface/graph-visualization/forces/center.js]: What about a description on the top about what this file is defining? Probably giving a name to the function would also help

File: js/components/interface/graph-visualization/forces/collide.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
+import { binarytree } from "d3-binarytree";
```
Comment (filippomc) [inline js/components/interface/graph-visualization/forces/collide.js]: What about a description of what's inside here? Looks like it's complex stuff!

Comment (filippomc) [reply js/components/interface/graph-visualization/forces/collide.js]: I'm starting to understand these are all forces we are applying, seems really important anyway to understand what they're there for

File: js/components/interface/graph-visualization/Graph.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    this.ggv.current.d3Force("charge", manyBodyForce())
+    this.ggv.current.d3Force("link", linkForce(data.links))
+    this.ggv.current.d3Force("center", centerForce())
+    this.ggv.current.d3Force('hold', holdForce())
+    if (url) {
+      this.addToScene()
+    } else {
+      this.zoomCameraToFitScene()
+    }
+    this.setState({ width, height })
```
[the remark this answers is not in the document]
Comment (rodriguez-facundo) [reply js/components/interface/graph-visualization/Graph.js]: Removed 👍

File: js/components/interface/graph-visualization/Graph.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+import linkForce from './forces/link'
+import holdForce from './forces/hold'
+import centerForce from './forces/center'
+import manyBodyForce from './forces/manyBody'
+
+const fullSizeStyle = { width: '100%', height: '100%' }
+
+export default class GeppettoGraphVisualization extends Component {
+  state = {
+    width: 0,
```
[the remark this answers is not in the document]
Comment (rodriguez-facundo) [reply js/components/interface/graph-visualization/Graph.js]: Done 👍

File: js/components/interface/graph-visualization/Graph.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    this.setState({ nodeSize: cameraZ / cameraSizeRatioToNodeSize })
+  }
+
+  render () {
+    const { width, height, nodeSize } = this.state
+    const { data, style, classes, ...others } = this.props;
+    
+    return (
+      <div style={{ ...fullSizeStyle, ...style, position:'relative' }}>
+        <div style={{ ...fullSizeStyle, position:'absolute' }} ref={this.container}>
```
[the remark this answers is not in the document]
Comment (rodriguez-facundo) [reply js/components/interface/graph-visualization/Graph.js]: All gone now

File: js/components/interface/graph-visualization/README.md (hunk 1 above)
Comment (rodriguez-facundo) [reply js/components/interface/graph-visualization/README.md]: Removed it

File: js/components/interface/graph-visualization/forces/center.js (hunk 2 above)
Comment (filippomc) [reply js/components/interface/graph-visualization/forces/center.js]: I mean just some comment on the top of the file to understand why they are there

**Pull request merged**